### ISSUE TITLE
Can I use the Grasshopper Application to work in the model?

### ISSUE THREAD

**Issue by Karbofo7** [LetsConstructIT/Grasshopper-TeklaDrawingLink#66]
Hi!
I'm trying to use the Grasshopper App in a model, but I get several errors:
Warning: [ Bake to Tekla component ] Input parameter Component failed to collect data
Error: [ Sort by Vector component ] Solution exception:The sequence contains no elements
Error: [ Sort by Vector component ] Solution exception:The sequence contains no elements
Error: [ Sort by Vector component ] Solution exception:The sequence contains no elements

If I run the script through a fully open Grasshopper - everything works. But if I run the script through the Grasshopper Application, I get these errors🤔
Unfortunately, I can't use the Grasshopper Component for my script "Transferring axis grids from AutoCAD to Tekla", because the Grasshopper Component always requires specifying an insertion point. Because of specifying an insertion point, the entire script breaks. The axis grids are created at the specified point, and not where they should be according to the dwg file.
Therefore, I try to use the Grasshopper Application, but it also gives me errors (

I will be very grateful if someone can help me run the script not through Rhino and Grasshopper, but through the Grasshopper Application or the Grasshopper Component

I attached my script and dwg file to the message, maybe it will be easier for you to understand the Grasshopper Application errors.

[Transferring axis grids from AutoCAD to Tekla.zip](https://github.com/user-attachments/files/17950344/Transferring.axis.grids.from.AutoCAD.to.Tekla.zip)

**Comment by olszewski-grzegorz** [comment]
Hello @Karbofo7 !
Thanks for sharing the script and sample DWG. Based on that I've tried Grasshopper Aplpication 1.3 and ... it worked.

I could only once get same errors as you - when I provided path to non-existent file. Please confirm, that you provide existing file.

Video on my side:

https://github.com/user-attachments/assets/13f47f12-dcac-4812-bccb-ec169208820b

**Comment by Karbofo7** [comment]
> Привет@Karbofo7! Спасибо за то, что поделились скриптом и образцом DWG. Исходя из этого, я попробовал Grasshopper Aplpication 1.3 и... это сработало.
> 
> Я смог получить те же ошибки, что и вы, только один раз - когда я указал путь к несуществующему файлу. Пожалуйста, подтвердите, что вы указали существующий файл.
> 
> Видео с моей стороны:
> 
>  Сетки.из.AutoCAD.mp4

Thank you! I will continue testing this script. Maybe it's because I have Tekla 2024🤔 Thank you so much for taking the time to check this error! I will try to figure out why it doesn't work on my computer

**Comment by Karbofo7** [comment]
> Hello @Karbofo7 ! Thanks for sharing the script and sample DWG. Based on that I've tried Grasshopper Aplpication 1.3 and ... it worked.
> 
> I could only once get same errors as you - when I provided path to non-existent file. Please confirm, that you provide existing file.
> 
> Video on my side:
> 
>  Grids.from.AutoCAD.mp4

Thank you! I will continue testing this script. Maybe it's because I have Tekla 2024🤔 Thank you so much for taking the time to check this error! I will try to figure out why it doesn't work on my computer

**Comment by olszewski-grzegorz** [comment]
I've checked with Tekla 2024, SP4.1, it works. 

You can try different approach with original Grasshopper Component. At the very beginning of your script, before placing objects in Tekla you could try to grab coordination system and try some transformations. Could be hard to figure out exactly what is needed but just a tip.
![image](https://github.com/user-attachments/assets/61c3d8f2-9ade-431f-bda2-5710c057c38b)

**Comment by Karbofo7** [comment]
> I've checked with Tekla 2024, SP4.1, it works.

Thank you. I tried to run this script in Tekla 2024 sp5 and in Tekla 2022 sp6. But Grasshopper Application really does not see the dwg file. You are right. I tried to make the path to the files only in English, I also moved the files to the root folder C:\ to avoid problems... Of course, I named both gh and dwg files in English, because Tekla does not like names in Cyrillic... But it still does not work for me🤔 I am very surprised that everything works perfectly for you exactly as I dreamed) Maybe I can somehow save a more detailed error log? To understand why Grasshopper Application does not want to read the dwg file on my computer, although it works for you.

**Comment by Karbofo7** [comment]
Forgive me, olszewski-grzegorz! I made a very stupid mistake😅 I cleaned and resaved my dwg file in dwg 2010 format, now everything works! I remembered that I had problems with newer dwg files in Tekla before. I am very ashamed that I took up your time😕 You are a wonderful person and a very cool developer! Thank you very much for your add-ons and plugins! They allow me to automate many routine processes. And I am also very grateful that you talked in detail about all your Grasshopper Tekla Drawing Link components in the Tekla Fundamentals Training lessons. I am glad that you cooperate with Krzysztof Wojslaw.
